File: Braintree/src/main/java/com/braintreepayments/api/models/KountConfiguration.java
package com.braintreepayments.api.models;

import android.text.TextUtils;

import com.braintreepayments.api.Json;

import org.json.JSONObject;

/**
 * Contains the remote Kount configuration for the Braintree SDK.
 */
public class KountConfiguration {

    private final static String KOUNT_MERCHANT_ID_KEY = "kountMerchantId";

    private String mKountMerchantId;

    /**
     * Parse a {@link KountConfiguration} from json.
     *
     * @param json The {@link JSONObject} to parse.
     * @return a {@link KountConfiguration} instance with the data that was able to be parsed from
     * the {@link JSONObject}.
     */
    public static KountConfiguration fromJson(JSONObject json) {
        if (json == null) {
            json = new JSONObject();
        }

        KountConfiguration kountConfiguration = new KountConfiguration();
        kountConfiguration.mKountMerchantId = Json.optString(json, KOUNT_MERCHANT_ID_KEY, "");

        return kountConfiguration;
    }

    /**
     * @return {@code true} if Kount is enabled, {@code false} otherwise.
     */
    public boolean isEnabled() {
        return !TextUtils.isEmpty(mKountMerchantId);
    }

    /**
     * @return the Kount Merchant Id.
     */
    public String getKountMerchantId() {
        return mKountMerchantId;
    }
}

File: BraintreeCore/src/main/java/com/braintreepayments/api/KountConfiguration.kt
package com.braintreepayments.api

import android.text.TextUtils
import org.json.JSONObject

/**
 * Contains the remote Kount configuration for the Braintree SDK.
 * @property kountMerchantId the Kount Merchant Id.
 * @property isEnabled `true` if Kount is enabled, `false` otherwise.
 */
internal data class KountConfiguration internal constructor(val kountMerchantId: String) {

    constructor(json: JSONObject?) : this(Json.optString(json, KOUNT_MERCHANT_ID_KEY, ""))

    val isEnabled: Boolean = !TextUtils.isEmpty(kountMerchantId)

    companion object {
        private const val KOUNT_MERCHANT_ID_KEY = "kountMerchantId"
    }
}